In [1]:
# Import Libraries
import pandas as pd
import pickle

In [2]:
# Load trained pipeline (preprocess + model)
with open("rf_baseline_pipeline.pkl", "rb") as f:
    model = pickle.load(f)

In [3]:
data_inf = {
    "hotel": ["Resort Hotel", "City Hotel"],
    "lead_time": [10, 120],
    "arrival_date_year": [2016, 2017],
    "arrival_date_month": ["July", "May"],
    "arrival_date_week_number": [27, 20],
    "arrival_date_day_of_month": [15, 25],
    "stays_in_weekend_nights": [2, 0],
    "stays_in_week_nights": [3, 2],
    "adults": [2, 2],
    "children": [0, 1],
    "babies": [0, 0],
    "meal": ["BB", "BB"],
    "country": ["PRT", "GBR"],
    "market_segment": ["Online TA", "Direct"],
    "distribution_channel": ["TA/TO", "Direct"],
    "is_repeated_guest": [0, 1],
    "previous_cancellations": [0, 1],
    "previous_bookings_not_canceled": [0, 2],
    "reserved_room_type": ["A", "D"],
    "assigned_room_type": ["A", "A"],
    "booking_changes": [0, 1],
    "deposit_type": ["No Deposit", "No Deposit"],
    "agent": ["9", "240"],
    "company": ["0", "0"],
    "days_in_waiting_list": [0, 0],
    "customer_type": ["Transient", "Contract"],
    "adr": [95.0, 120.0],
    "required_car_parking_spaces": [0, 1],
    "total_of_special_requests": [1, 0]
}

df_inf = pd.DataFrame(data_inf)
df_inf

,hotel,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,...,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests
0,Resort Hotel,10,2016,July,27,15,2,3,2,0,...,A,0,No Deposit,9,0,0,Transient,95.0,0,1
1,City Hotel,120,2017,May,20,25,0,2,2,1,...,A,1,No Deposit,240,0,0,Contract,120.0,1,0


In [4]:
# Predict class
pred_class = model.predict(df_inf)

# Predict probability
pred_prob = model.predict_proba(df_inf)[:, 1]

In [5]:
results = pd.DataFrame({
    "Predicted_Class": pred_class,
    "Probability_Cancel": pred_prob
})

results["Interpretation"] = results["Predicted_Class"].apply(
    lambda x: "Akan Cancel" if x == 1 else "Tidak Akan Cancel"
)

results

,Predicted_Class,Probability_Cancel,Interpretation
0,0,0.405,Tidak Akan Cancel
1,0,0.150,Tidak Akan Cancel


## Kesimpulan Inference Model

Berdasarkan hasil inference menggunakan model Random Forest, diperoleh kesimpulan sebagai berikut:

1. **User 1** diprediksi **tidak akan melakukan pembatalan reservasi** dengan probabilitas pembatalan sebesar **40.5%**.  
   Nilai probabilitas ini menunjukkan bahwa meskipun model mengklasifikasikan reservasi sebagai *tidak cancel*, terdapat **tingkat risiko yang cukup moderat**. Karakteristik User 1 berada relatif dekat dengan batas keputusan model, sehingga masih memiliki potensi pembatalan yang perlu diwaspadai dari perspektif bisnis.

2. **User 2** diprediksi **tidak akan melakukan pembatalan reservasi** dengan probabilitas pembatalan sebesar **15.0%**.  
   Probabilitas yang relatif rendah ini mengindikasikan bahwa User 2 memiliki karakteristik yang **lebih konsisten dengan kelompok non-cancel**. Model tidak menangkap sinyal perilaku yang kuat yang mengarah pada risiko pembatalan.

### Ringkasan Interpretasi

- Kedua reservasi diprediksi **tidak akan dibatalkan**
- Namun, **tingkat keyakinan model berbeda**
  - User 1 memiliki risiko pembatalan yang lebih tinggi dan perlu perhatian
  - User 2 memiliki risiko pembatalan yang rendah dan relatif aman
- Informasi probabilitas ini dapat dimanfaatkan untuk:
  - Penentuan strategi overbooking
  - Prioritisasi konfirmasi reservasi
  - Intervensi preventif pada reservasi dengan risiko menengah